src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs

```diff
@@ -509,7 +536,9 @@ internal void StartOperationCommon(Socket? socket, SocketAsyncOperation operatio
             _currentSocket = socket;
 
             // Capture execution context if needed (it is unless explicitly disabled).
-            if (_flowExecutionContext)
+            // If Telemetry is enabled, make sure to capture the context if we're making a Connect or Accept call to preserve the activity
+            if (_flowExecutionContext ||
```

Comment:
We have call sites like
```c#
// Potentially
TelemetryStart(endPoint/address...);

e.StartOperationCommon(...);
e.StartOperationX(...);
```

`StartOperationCommon` does not have enough information to decide whether to log the Start event. For example we don't want to log `ConnectStart` for the parent multiConnect operation, but we don't have that info available in the method - we set the `_multipleConnect` field in `e.StartOperationConnect(...)`.

Even if we did, the logic here would get more complicated to perform the checks (where as it's more obvious what's happening at the call sites themselves). I tried to keep the product change as small as possible.

Alternatively we could introduce an argument to indicate whether to ignore the `_flowExecutionContext` flag to avoid the duplication you mentioned. I don't have a strong opinion on this.